### PR TITLE
improvements

### PR DESCRIPTION
**Pull request by jvmakine**

### CODE REVIEW

File: parser.go [hunk 1]
```diff
@@ -82,14 +83,14 @@ type Extensions struct {
 type Reserved struct {
 	Pos lexer.Position
 
-	Reserved []Range `"reserved" @@ { "," @@ }`
+	Ranges     []Range  `( @@ { "," @@ }`
```
Comment (alecthomas) [inline parser.go]: `()` is redundant within a struct - there's an implicit group

File: parser.go [hunk 2]
```diff
@@ -82,14 +83,14 @@ type Extensions struct {
...
+	FieldNames []string `  | @String { "," @String } )`
 }
 
 type Range struct {
-	Ident string `  @String`
-	Start int    `| ( @Int`
-	End   *int   `  [ "to" ( @Int`
-	Max   bool   `           | @"max" ) ] )`
+	Start int  `( @Int`
```
Comment (alecthomas) [inline parser.go]: Same here.

File: parser.go [hunk 3]
```diff
@@ -278,11 +279,28 @@ type MapType struct {
...
 // Parse protobuf.
 func Parse(filename string, r io.Reader) (*Proto, error) {
 	p := &Proto{}
+
+	l := lexer.MustSimple([]lexer.Rule{
+		{"String", `"(\\"|[^"])*"|'(\\'|[^'])*'`, nil},
+		{"Ident", `[a-zA-Z_]([a-zA-Z_0-9])*`, nil},
+		{"Float", `[-+]?((\d*\.\d+)|inf)`, nil},
+		{"Int", `(0[xX][0-9A-Fa-f]+)|([-+]?\d+)`, nil},
+		{"Whitespace", `[ \t\n\r]+`, nil},
```
Comment (alecthomas) [inline parser.go]: `\s`?

File: parser.go [hunk 4]
```diff
@@ -278,11 +279,28 @@ type MapType struct {
...
+		{"BlockComment", `/\*([^*]|[\r\n]|(\*+([^*/]|[\r\n])))*\*+/`, nil},
+		{"LineComment", `//(.*)[^\n]*\n`, nil},
+		{".", `.`, nil},
```
Comment (alecthomas) [inline parser.go]: What's this for?

**Review by alecthomas — APPROVED** [review]

File: parser.go (hunk 4 above)
Comment (jvmakine) [reply parser.go]: That was me being lazy and capturing everything in the lexer as a symbol. Will replace with a list of supported symbols.

**Pull request merged**